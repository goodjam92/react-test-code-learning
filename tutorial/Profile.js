const Profile = ({ userName, name }) => {
  return (
    <div>
      <b>{userName}</b>
      <span>({name})</span>
    </div>
  );
};

export default Profile;
