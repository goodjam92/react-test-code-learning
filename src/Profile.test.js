import { mount } from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches sanpshot", () => {
    const wrapper = mount(<Profile userName={"goodjam"} name={"JAEMO KOO"} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders userName and name", () => {
    const wrapper = mount(<Profile userName={"goodjam"} name={"JAEMO KOO"} />);
    expect(wrapper.props().userName).toBe("goodjam");
    expect(wrapper.props().name).toBe("JAEMO KOO");

    // DOM에 우리가 원하는 텍스트가 나타나는지 확인하는 테스트 코드
    const boldElement = wrapper.find("b");
    expect(boldElement.contains("goodjam")).toBe(true);
    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("(JAEMO KOO)");
  });
});
