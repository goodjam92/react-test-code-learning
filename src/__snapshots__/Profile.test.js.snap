// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches sanpshot 1`] = `
<Profile
  name="JAEMO KOO"
  userName="goodjam"
>
  <div>
    <b>
      goodjam
    </b>
    <span>
      (
      JAEMO KOO
      )
    </span>
  </div>
</Profile>
`;
